tests/test_animate.py

```diff
@@ -25,6 +25,17 @@ def check_animate_output(
     assert captured.err[:chars_to_test] == expected_string[:chars_to_test]
 
 
+def test_delay_suppresses_output(capsys, monkeypatch):
+    monkeypatch.setattr(pipx.animate, "stderr_is_tty", True)
+
+    test_string = "asdf"
+
+    with pipx.animate.animate(test_string, do_animation=True, delay=0.1):
```

Comment:
I think you might need more time buffer here.  I started out with only 0.1 extra time in `check_animate_output`, and occasionally (especially on Travis' Windows testers) I would get spurious failures.  I think you may need a `delay` of a few tenths of a second more than the `sleep()` time at least.